dhall/src/Dhall/Pretty/Internal.hs

```diff
@@ -1410,13 +1410,13 @@ prettyPrinters characterSet =
                                     <>  prettyValue val
                                     )
           where
-            (preSeparator, preComment) =
-                case key of
-                    (_, _, mSrc2) :| [] | not (containsComment mSrc2) ->
-                        (" ", Pretty.hardline <> "    ")
-                    _ ->
-                        (Pretty.hardline, " ")
-
+            (preSeparator, preComment)
+                | not (any hasComment key) = 
+                    (" ", Pretty.hardline <> "    ")
```

Comment:
With this rule we get

```
{   x
  . {- bla -}
    y = z
}
```

which IMHO somewhat "hides" the value in the last path segment.

I think the rule should be that any comments except one preceding the very first segment should trigger the line break before the separator.